### ISSUE TITLE
cmd/validate: Prevalidate AMI Name

### ISSUE THREAD

**Issue by craigbarrau** [hashicorp/packer#4761]
Steps to reproduce:
1. create the file `bug.json`
```
{
  "variables": {
    "vm_name": "Centos 7 + Stuff"
  },
  "builders": [
    {
      "type": "amazon-ebs",
      "region": "us-west-2",
      "source_ami": "ami-d2c924b2",
      "instance_type": "m3.medium",
      "ssh_username": "centos",
      "ssh_pty": "true",
      "ami_name": "{{user `vm_name`}} {{timestamp}} - Testing Delete Me",
      "ami_description": "Built by Packer"
    }
  ]
}
```
2. Run `packer build bug.json` (assuming AWS CLI is established)
3. Observe the logs. Notice it validates the AMI name but doesn't fail on that, it fails much later after the instance has been created:
```
amazon-ebs output will be in this color.

==> amazon-ebs: Prevalidating AMI Name...
    amazon-ebs: Found Image ID: ami-d2c924b2
==> amazon-ebs: Creating temporary keypair: OMITTED
==> amazon-ebs: Launching a source AWS instance...
    amazon-ebs: Instance ID: OMITTED
==> amazon-ebs: Waiting for instance (OMITTED) to become ready...
==> amazon-ebs: Adding tags to source instance
==> amazon-ebs: Waiting for SSH to become available...
==> amazon-ebs: Connected to SSH!
==> amazon-ebs: Stopping the source instance...
==> amazon-ebs: Waiting for the instance to stop...
==> amazon-ebs: Creating the AMI: Centos 7 + Stuff 1491348761 - Testing Delete Me
==> amazon-ebs: Error creating AMI: InvalidAMIName.Malformed: AMI names must be between 3 and 128 characters long, and may contain letters, numbers, '(', ')', '.', '-', '/' and '_'
==> amazon-ebs: 	status code: 400, request id: OMITTED
==> amazon-ebs: Terminating the source AWS instance...
==> amazon-ebs: Cleaning up any extra volumes...
==> amazon-ebs: No volumes to clean up, skipping
==> amazon-ebs: Deleting temporary keypair...
Build 'amazon-ebs' errored: Error creating AMI: InvalidAMIName.Malformed: AMI names must be between 3 and 128 characters long, and may contain letters, numbers, '(', ')', '.', '-', '/' and '_'
	status code: 400, request id: OMITTED
```

**Comment by mwhooker** [comment]
Thanks for the report. 

This would be a good enhancement for anyone looking to start contributing to packer. You'd want to add that check here https://github.com/hashicorp/packer/blob/master/builder/amazon/common/ami_config.go#L70 and test for it here https://github.com/hashicorp/packer/blob/master/builder/amazon/common/ami_config_test.go

**Comment by craigbarrau** [comment]
Thanks @mwhooker... I'll have a crack at that.

**Comment by craigbarrau** [comment]
My first contribution: https://github.com/hashicorp/packer/pull/4762
Hope I went ok :)

**Comment by mwhooker** [comment]
That's awesome, thanks!

**Comment by mwhooker** [comment]
closed in #4762
